src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -411,12 +420,25 @@ public FailFastRunner(Runner wrappedRunner) {
     this.useExperimentalRunner = useExperimentalRunner;
   }
 
+  private void setTestsToRun(Collection<String> tests) {
```

Comment:
It's not clear why this was broken out. Adding a comment that explains "why mutable state" for this case would be good.